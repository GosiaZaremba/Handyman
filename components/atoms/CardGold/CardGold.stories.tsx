import React from 'react';
import {storiesOf} from '@storybook/react-native';
import {CardGold} from './CardGold';
import {action} from '@storybook/addon-actions';
import {text} from '@storybook/addon-knobs';

storiesOf('atoms/CardGold', module)
  .addDecorator(story => story())
  .add('default', () => <CardGold children={''} />);
