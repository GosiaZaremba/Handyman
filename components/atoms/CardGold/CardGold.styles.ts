import {StyleSheet} from 'react-native';
import {Colors} from '../../../constants/colors';

export default StyleSheet.create({
  container: {
    backgroundColor: Colors.gold.dark,
    borderRadius: 5,
    width: '100%',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
});
