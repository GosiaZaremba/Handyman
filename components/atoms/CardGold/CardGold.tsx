import React from 'react';
import {View} from 'react-native';
import styles from './CardGold.styles';

export type Props = {
  children: any;
};

export const CardGold: React.FC<Props> = ({children}) => {
  return <View style={styles.container}>{children}</View>;
};
