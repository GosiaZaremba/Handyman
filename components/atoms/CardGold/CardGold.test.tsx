import React from 'react';
import {render} from '@testing-library/react-native';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import {CardGold} from './CardGold';
import {View} from 'react-native';

describe('Atom CardGold', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <CardGold testID="atom-cardGold">
          <View />
        </CardGold>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure children are rendered correctly', () => {
    const {getByTestId} = render(
      <CardGold testID="atom-cardGold">
        <View />
      </CardGold>,
    );
    const cardGold = getByTestId('atom-cardGold');
    expect(cardGold.props.children.type.displayName).toBe('View');
  });
});
