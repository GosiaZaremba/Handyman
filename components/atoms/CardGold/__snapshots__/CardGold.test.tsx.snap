// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom CardGold Component Snapshot 1`] = `
<BVLinearGradient
  colors={
    Array [
      4294945379,
      4294938123,
    ]
  }
  endPoint={
    Object {
      "x": 0,
      "y": 0.5,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 1,
      "y": 0.5,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ff8e0b",
      "borderRadius": 5,
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
  testID="atom-cardGold"
>
  <View />
</BVLinearGradient>
`;
